specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2025-01-15-preview/cluster.json

```diff
@@ -0,0 +1,2331 @@
+            "RSA-OAEP-256",
+            "RSA1_5"
+          ],
+          "x-ms-enum": {
+            "name": "JsonWebKeyEncryptionAlgorithm",
+            "modelAsString": true
+          }
+        },
+        "msiResourceId": {
+          "type": "string",
+          "description": "Resource ID of Managed Identity that is used to access the key vault."
+        },
+        "encryptionAtHost": {
+          "type": "boolean",
+          "default": false,
+          "description": "Indicates whether or not resource disk encryption is enabled."
+        }
+      }
+    },
+    "EncryptionInTransitProperties": {
+      "type": "object",
+      "description": "The encryption-in-transit properties.",
+      "properties": {
+        "isEncryptionInTransitEnabled": {
+          "type": "boolean",
+          "default": false,
+          "description": "Indicates whether or not inter cluster node communication is encrypted in transit."
+        }
+      }
+    },
+    "UpdateGatewaySettingsParameters": {
+      "type": "object",
+      "properties": {
+        "restAuthCredential.isEnabled": {
+          "type": "boolean",
+          "default": true,
+          "x-ms-client-name": "isCredentialEnabled",
+          "description": "Indicates whether or not the gateway settings based authorization is enabled."
+        },
+        "restAuthCredential.username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "description": "The gateway settings user name."
+        },
+        "restAuthCredential.password": {
+          "type": "string",
+          "x-ms-client-name": "password",
+          "description": "The gateway settings user password."
+        },
+        "restAuthEntraUsers": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EntraUserInfo",
+            "description": "Details of an Entra user for gateway access."
+          },
+          "description": "List of Entra users for gateway access."
+        }
+      },
+      "description": "The update gateway settings request parameters. Note either basic or entra user should be provided at a time."
+    },
+    "GatewaySettings": {
+      "type": "object",
+      "properties": {
+        "restAuthCredential.isEnabled": {
+          "type": "string",
+          "x-ms-client-name": "isCredentialEnabled",
+          "description": "Indicates whether or not the gateway settings based authorization is enabled.",
+          "readOnly": true
+        },
+        "restAuthCredential.username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "description": "The gateway settings user name.",
+          "readOnly": true
+        },
+        "restAuthCredential.password": {
+          "type": "string",
+          "x-ms-client-name": "password",
+          "description": "The gateway settings user password.",
+          "readOnly": true
+        },
+        "restAuthEntraUsers": {
```

Comment:
Please check if these two are need for restAuthEntraUsers
  "x-ms-client-name": "",
          "readOnly":

Reply:
"x-ms-client-name": As per documentation it is an alternative name of the property when generating the client sdk. In our case  key values are proper (camel case) and do not require this attribute.
"readOnly": Entra Users properties (objectId, displayName and upn) can be updated in UpdateGatewaySettingsParameters so we can not mark them read only for GatewaySettings.